### DISCUSSION TITLE
When and how often is conftest.py loaded?

### DISCUSSION THREAD

**Discussion by andrii-korotkov** [pytest-dev/pytest#11312] [Q&A] [answered]
Hello,
I have a use case where a conftest contains imports with context managers like
`
with mock.patch("common.library", mock_object):
    import mymodule
`
So that mymodule has common.library replaces with a mock. It can be pretty convenient when common.library is used in multiple places and same instance is passed around. With this kind of import it doesn't have to be mocked in all places manually.

I wonder if this import would happen once for all tests, once per test module, one per individual test? In other words, would the mock_object used be the same for all tests, all tests in a module or be separate for each test? This affects possible cleanups to be done.
Thank you.

**Comment by RonnyPfannschmidt** [comment, marked as answer]
first, a conftest should never throw in mocks at import time, it completely breaks details even if it looks like its working

second - conftests are either loaded at pytest startup (early conftests) or at collection time (the rest of them)

all conftests are supposed to be imported once
